models/favorites.js

```diff
@@ -7,5 +9,54 @@ module.exports = (sequelize, DataTypes) => {
     Favorites.belongsTo(models.User),
     Favorites.belongsTo(models.Location)
   };
+
+  Favorites.getLocations = function(userid){
+    var Favorites = this
+    var Location = require('../models').Location;
+    return new Promise(function (resolve, reject){
+      Favorites.findAll({
+        where: {UserId: userid}
+      })
+      .then(favorites => {
+        locationArray(favorites)
+        .then(locations => {
+          forecastArray(locations)
+          .then(forecast => {
+            resolve(forecast) })
+          .catch(error => {})
+        })
+        .catch(error => {});
+      })
+    })
+  };
+
+  function locationArray(favorites){
+    var Location = require('../models').Location;
+    return new Promise(function (resolve, reject){
+      var locationArray = [];
+      for (let i = 0; i < favorites.length; i++){
```

Comment:
If you're looking to refactor, this might be a situation where you could use a JavaScript map function. I haven't tried implementing it myself yet, but this article might be able to help you with the syntax: http://cryto.net/~joepie91/blog/2015/05/04/functional-programming-in-javascript-map-filter-reduce/

Reply:
You know i actually looked at something about that! i think you are right in that that would help clean that up quite a bit.